appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/jwt/JWTInjectableType.java

```diff
@@ -148,8 +148,7 @@ private void installCoreConverter() {
         } else if (coreClass.equals(Long.class)) {
             converter = e -> ((JsonNumber) e).longValue();
         } else if (coreClass.equals(Boolean.class)) {
-            // TODO
-            converter = e -> e;
+            converter = e -> convertToBoolean(e);
```

Comment:
I see that file already contain `Class` comparison using `equals`.
As it is not reasonable to have two equal instances of a `Class` unless they are in fact identical I would generally suggest to not shy away from using the `==` and `!=` operators when working with `Class`es.